src/pages/understand-stacks/running-api-node.md

```diff
@@ -0,0 +1,300 @@
+- [psql](http://postgresguide.com/utilities/psql.html) installed locally
+- [jq](https://stedolan.github.io/jq/download/)
+
+### Firewall configuration
+
+In order for the API node services to work correctly, you must configure any network firewall rules to allow traffic on
+the ports discussed in this section. The details of network and firewall configuration are highly specific to your
+machine and network, so a detailed example isn't provided.
+
+The following ports must open on the host machine:
+
+Ingress:
+
+- postgres (open to `localhost` only):
+  - `5432 TCP`
+- stacks-blockchain-api
+  - `3999 TCP`
+- stacks-blockchain (open to `0.0.0.0/0`):
+  - `20443 TCP`
+  - `20444 TCP`
+
+Egress:
+
+- `8332`
+- `8333`
+- `20443-20444`
+
+These egress ports are for syncing `stacks-blockchain` and Bitcoin headers. If they're not open, the sync will fail.
+
+## Step 1: initial setup
+
+In order to run the API node, you must download the Docker images and create a directory structure to hold the
+persistent data from the services. Download and configure the Docker images with the following commands:
+
+```sh
+docker pull blockstack/stacks-blockchain-api && docker pull blockstack/stacks-blockchain && docker pull postgres:alpine
+docker network create stacks-blockchain > /dev/null 2>&1
+```
+
+Create a directory structure for the service data with the following command:
+
+```sh
+mkdir -p ./stacks-node/{persistent-data/postgres,persistent-data/stacks-blockchain,bns,config} && cd stacks-node
+```
+
+## Step 2: running Postgres
+
+The `postgres:alpine` Docker container can be run with default settings. You must set the password for the user to
+`postgres` with the `POSTGRES_PASSWORD` environment variable. The following command starts the image:
+
+```sh
+docker run -d --rm \
+  --name postgres \
+  --net=stacks-blockchain \
+  -e POSTGRES_PASSWORD=postgres \
+  -v $(pwd)/persistent-data/postgres:/var/lib/postgresql/data \
+  -p 5432:5432
+  postgres:alpine
+```
+
+You can verify the running Postgres instance on port `5432` with the command
+
+```sh
+docker ps --filter name=postgres
+```
+
+## Step 3: running Stacks blockchain API
+
+The `stacks-blockchain-api` image requires several environment variables to be set. To reduce the complexity of the
+run command, you should create a new `.env` file and add the following to it using a text editor:
+
+```dotenv
+NODE_ENV=production
+GIT_TAG=master
+PG_HOST=postgres
+PG_PORT=5432
+PG_USER=postgres
+PG_PASSWORD=postgres
+PG_DATABASE=postgres
+STACKS_CHAIN_ID=0x00000001
+V2_POX_MIN_AMOUNT_USTX=90000000260
+STACKS_CORE_EVENT_PORT=3700
+STACKS_CORE_EVENT_HOST=0.0.0.0
+STACKS_BLOCKCHAIN_API_PORT=3999
+STACKS_BLOCKCHAIN_API_HOST=0.0.0.0
+STACKS_BLOCKCHAIN_API_DB=pg
+STACKS_CORE_RPC_HOST=stacks-blockchain
+STACKS_CORE_RPC_PORT=20443
+BNS_IMPORT_DIR=/bns-data
+```
+
+-> In this example, this guide configured the API to import BNS data with the `BNS_IMPORT_DIR` variable. To disable this
```

Comment:
This text sounds off to me. perhaps a change like `In this example, we have configured...` ?

Reply:
Using `we` goes against our style guide - How about `This guide configures the API...` ?

Reply:
understood. I'll defer to you on how it should be presented - the original text just flows a bit odd to me.